src/main/scala/viper/gobra/frontend/info/implementation/property/TypeMerging.scala

```diff
@@ -27,10 +26,10 @@ trait TypeMerging extends BaseProperty { this: TypeInfoImpl =>
 
           case (a, UNTYPED_INT_CONST) if underlyingType(a).isInstanceOf[IntT] => Some(a)
           case (UNTYPED_INT_CONST, b) if underlyingType(b).isInstanceOf[IntT] => Some(b)
-          case (a, UNTYPED_INT_CONST) if underlyingType(a).isInstanceOf[Float64T] => Some(a)
-          case (UNTYPED_INT_CONST, b) if underlyingType(b).isInstanceOf[Float64T] => Some(b)
-          case (a, UNTYPED_INT_CONST) if underlyingType(a).isInstanceOf[Float32T] => Some(a)
-          case (UNTYPED_INT_CONST, b) if underlyingType(b).isInstanceOf[Float32T] => Some(b)
+          case (a, UNTYPED_INT_CONST) if underlyingType(a).isInstanceOf[FloatT] => Some(a)
+          case (UNTYPED_INT_CONST, b) if underlyingType(b).isInstanceOf[FloatT] => Some(b)
+          case (a, UnboundedFloatT) if underlyingType(a).isInstanceOf[FloatT] => Some(a)
+          case (UnboundedFloatT, b) if underlyingType(b).isInstanceOf[FloatT] => Some(b)
```

Comment:
one of these rules implies that we can assign sth of a concrete float type to an unbounded float (literal). Is such a rule strictly necessary?

Reply:
The way I understand it, these rules express that if you have an unbounded float and a concrete float, these two types get merged to the concrete one, which is however not directly related to assignability